pgmpy/base/DAG.py

```
@@ -1301,20 +1335,56 @@ def get_random(
             )
         return dag
 
-    def to_graphviz(self):
+    def to_graphviz(self, plot_edge_strength=False):
         """
-        Retuns a pygraphviz object for the DAG. pygraphviz is useful for
+        Returns a pygraphviz object for the DAG. pygraphviz is useful for
         visualizing the network structure.
 
+        Parameters
+        ----------
+        plot_edge_strength: boolean (default: False)
+            Whether to plot edge strengths as labels on edges.
+            Requires edge strengths to be computed first using the `edge_strength` method.
+
+        Returns
+        -------
+        pygraphviz object: pygraphviz.AGraph object
+            pygraphviz object for the DAG.
+
         Examples
         --------
         >>> from pgmpy.utils import get_example_model
         >>> model = get_example_model('alarm')
         >>> model.to_graphviz()
         <AGraph <Swig Object of type 'Agraph_t *' at 0x7fdea4cde040>>
-        >>> model.draw('model.png', prog='neato')
         """
-        return nx.nx_agraph.to_agraph(self)
+        # Validate edge strengths if plotting is requested
+        if plot_edge_strength:
+            missing_strengths = []
+            for u, v in self.edges():
+                if "strength" not in self.edges[(u, v)]:
+                    missing_strengths.append((u, v))
+
+            if missing_strengths:
+                logger.warning(
```

Same, we should throw an error here.